lib/beautiful/colorful.lua

```diff
@@ -0,0 +1,1395 @@
+
+--- Return the decimal representation of the color including the alpha channel
+-- @treturn number The decimal represention of the color including the alpha
+-- channel
+-- @usage local deca_color = c.as:deca()
+function color.as:deca()
+    return math.floor(
+          lshift(round(self.red   * 0xFF), 24)
+        + lshift(round(self.green * 0xFF), 16)
+        + lshift(round(self.blue  * 0xFF), 8)
+        +        round(self.alpha * 0xFF)
+    )
+end
+
+--- Return the decimal representation of the color
+-- @treturn number The decimal represention of the color not including the
+-- alpha channel
+-- @usage local dec_color = c.as:dec()
+function color.as:dec()
+    return math.floor(
+          lshift(round(self.red   * 0xFF), 16)
+        + lshift(round(self.green * 0xFF), 8)
+        +        round(self.blue  * 0xFF)
+    )
+end
+
+-- }}} Decimal output
+
+-- {{{ Multi-return output
+
+--- Return the Hue, Saturation, Lightness, and Alpha
+-- @treturn number Hue [0,360]
+-- @treturn number Saturation [0,1]
+-- @treturn number Lightness [0,1]
+-- @treturn number Alpha [0,1]
+-- @usage local h,s,l,a = c.as:hsla()
+function color.as:hsla()
+    return unpack({
+        self.hue,
+        self.saturation,
+        self.lightness,
+        self.alpha
+    })
+end
+
+--- Return the Hue, Saturation, and Lightness
+-- @treturn number Hue [0,360]
+-- @treturn number Saturation [0,1]
+-- @treturn number Lightness [0,1]
+-- @usage local h,s,l = c.as:hsl()
+function color.as:hsl()
+    return unpack({
+        self.hue,
+        self.saturation,
+        self.lightness,
+    })
+end
+
+--- Return the Red, Green, Blue, and Alpha
+-- @treturn number Red [0,1]
+-- @treturn number Green [0,1]
+-- @treturn number Blue [0,1]
+-- @treturn number Alpha [0,1]
+-- @usage local r,g,b,a = c.as:rgba()
+function color.as:rgba()
+    return unpack({
+        self.red,
+        self.green,
+        self.blue,
+        self.alpha
+    })
+end
+
+--- Return the Red, Green, and Blue
+-- @treturn number Red [0,1]
+-- @treturn number Green [0,1]
+-- @treturn number Blue [0,1]
+-- @usage local r,g,b = c.as:rgb()
+function color.as:rgb()
+    return unpack({
+        self.red,
+        self.green,
+        self.blue,
+    })
+end
+
+-- }}} Multi-return output
+
+-- @section end
+
+-- }}} Color output functions
+
+--- Color metamethods
+-- @section color_meta
+
+color_mt.__index = color
+color_mt.__tostring = color.as.hex
+
+--- Shorthand invert of a color
+-- @function color.__unm
+-- @see beautiful.colorful.color.invert
+color_mt.__unm = color.invert
+
+--- Add the RGBA values of two colors
+-- @function color.__add
+-- @tparam lhs beautiful.colorful.color A color object
+-- @tparam rhs beautiful.colorful.color A color object
+-- @treturn beautiful.colorful.color A new color object with RGBA values added
+-- @usage colorful.color("#123") + colorful.color("#321") -- returns #444444
+color_mt.__add = function(lhs, rhs)
+    if type(rhs) ~= "table" then return nil end
```

Comment:
Would `error` be more appropriate than `nil`? Actually... how about just dropping this check? There are enough tables which still cannot be added, so why special case this?

Reply:
It's a simple check just to make sure you're not adding a number or string.